### PR TITLE
HTTPSE: use cpp code from laptop/android.

### PR DESCRIPTION
**Pull request by garvankeeley**
- use alternate JSON parser, original code uses chromium lib
- use c++11 regex, original code uses 3rd party lib

Original: chromium/chrome/browser/net/blockers/blockers_worker.cc

Part of https://github.com/brave/browser-ios/issues/310

### CODE REVIEW

**Comment by garvankeeley** [comment]
@SergeyZhukovsky r?

**Comment by garvankeeley** [comment]
@diegodoval not sure I'll land this on master until we have done more performance work. 
I removed all the caches that I have in the swift version, I can add in cpp caches, but we have bigger problems to deal with first

File: brave/src/webfilters/HttpsEverywhereCpp.mm [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        domain_to_check.insert(0, ".");
+        domain_to_check.insert(0, domains[i]);
+        std::string prefix;
+        if (i > 0) {
+            prefix = "*.";
+        }
+        query += prefix + domain_to_check + "'";
+    }
+
+    sqlite3_profile(httpse_db_, xProfile, NULL);
```
Comment (garvankeeley) [inline brave/src/webfilters/HttpsEverywhereCpp.mm]: needs to be off before landing

**Comment by diegodoval** [comment]
change itself looks ok -- not sure this is the way to go though.

**Comment by garvankeeley** [comment]
concerns ?

File: brave/src/webfilters/HttpsEverywhereCpp.mm [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    if (err != SQLITE_OK) {
+        // std::cout << "sqlite db open error " << dbFilePath.c_str() << ", err == " << err;
+        return false;
+    }
+
+    return true;
+}
+
+std::string applyHTTPSRule(const std::string& originalUrl, const std::string& rule) {
+    auto json_object = json::parse(rule);
```
Comment (garvankeeley) [inline brave/src/webfilters/HttpsEverywhereCpp.mm]: @bridiver applyHTTPSRule shows how the JSON is used

**Pull request closed without merging**